compiler/qsc_partial_eval/tests/branching.rs

```diff
@@ -236,41 +235,77 @@ fn if_expression_with_dynamic_condition() {
 
     // Verify the callables added to the program.
     let mresetz_callable_id = CallableId(1);
-    assert_callable(&program, mresetz_callable_id, &mresetz_callable());
+    assert_callable(
+        &program,
+        mresetz_callable_id,
+        &expect![[r#"
+        Callable:
+            name: __quantum__qis__mresetz__body
+            call_type: Measurement
+            input_type:
+                [0]: Qubit
+                [1]: Result
+            output_type: <VOID>
+            body: <NONE>"#]],
+    );
     let read_result_callable_id = CallableId(2);
-    assert_callable(&program, read_result_callable_id, &read_result_callable());
+    assert_callable(
+        &program,
+        read_result_callable_id,
+        &expect![[r#"
+        Callable:
+            name: __quantum__qis__read_result__body
+            call_type: Readout
+            input_type:
+                [0]: Result
+            output_type: Boolean
+            body: <NONE>"#]],
+    );
     let op_a_callable_id = CallableId(3);
-    assert_callable(&program, op_a_callable_id, &single_qubit_intrinsic_op_a());
+    assert_callable(
+        &program,
+        op_a_callable_id,
+        &expect![[r#"
+        Callable:
+            name: opA
+            call_type: Regular
+            input_type:
+                [0]: Qubit
+            output_type: <VOID>
+            body: <NONE>"#]],
+    );
 
     // Set the IDs of the blocks we want to verify.
     let initial_block_id = BlockId(0);
     let continuation_block_id = BlockId(1);
     let if_block_id = BlockId(2);
 
     // Verify the branch instruction in the initial-block.
-    let condition_var = Variable {
-        variable_id: 1.into(),
-        ty: Ty::Boolean,
-    };
-    let branch_inst = Instruction::Branch(condition_var, if_block_id, continuation_block_id);
-    assert_block_last_instruction(&program, initial_block_id, &branch_inst);
+    assert_block_last_instruction(
```

Comment:
For branching tests, I think it is more readable to have just an `assert_blocks` function that compares against all the blocks than asserting each one of the blocks individually. IMO, it is easier to see where the branch and jump instructions point to if all the blocks are displayed at once.

Reply:
Makes sense, I can add that!